src/index.js

```diff
@@ -1,25 +1,156 @@
 import './styles/index.css';
+import Task from './Task';
 
-const task = [
-  {
-    index: 1,
-    description: 'Wash the dishes',
-    completed: 'true',
-  },
-  {
-    index: 2,
-    description: 'Complete To Do list project',
-    completed: 'true',
-  },
-  {
-    index: 3,
-    description: 'Prepare dinner',
-    completed: 'true',
-  },
-];
-
-const list = task.map((item) => `
- <li><input type ="checkbox">${item.description}</li>`).join('');
-
+const listContainer = document.querySelector('.list-container');
+const inputField = document.querySelector('input');
 const tasks = document.querySelector('.tasks');
-tasks.innerHTML += list;
\ No newline at end of file
+const button = document.querySelector('button');
+
+// add task to the UI functionality
+const listArray = [];
+
+const addTask = (taskValue) => {
+  const listItem = document.createElement('li');
+  listItem.innerHTML = `
+  <input type="checkbox" class="checkbox"/>
+  <span>${taskValue}</span >
+  <i class="fa-solid fa-ellipsis-vertical edit-btn"></i>
+  <i class="fa-solid fa-trash-can delete-btn"></i>
+  `;
+  tasks.appendChild(listItem);
+
+  // checkbox functionality
+  const checkbox = document.querySelectorAll('.checkbox');
+  checkbox.forEach((i) => {
+    i.addEventListener('click', () => {
+      i.parentElement.classList.toggle('checkContainer');
+      i.nextElementSibling.classList.toggle('checkTask');
+      i.parentElement.lastElementChild.classList.toggle('active');
+      i.parentElement.lastElementChild.previousElementSibling.classList.toggle('disable');
+      updateStorage();
+    });
+  });
+
+  // Add data to local storage
+  const store = new Task(taskValue, false, checkbox.length - 1);
+  listArray.push(store);
+  localStorage.setItem('list', JSON.stringify(listArray));
+
+  // Edit event listener
+  const editBtn = document.querySelectorAll('.edit-btn');
+  editBtn.forEach((i) => {
+    i.addEventListener('click', () => {
+      editTask(listItem, i.previousElementSibling);
+      updateStorage();
+    });
+  });
+
+  //  Delete event listener
+  const removeBtn = document.querySelectorAll('.delete-btn');
+  removeBtn.forEach((i) => {
+    i.addEventListener('click', () => {
+      deleteTask(i.parentElement);
+    });
+  });
+};
```

Comment:
- [ ] Adding and removing tasks from the list is working fine but there is a little issue, After adding a task the index values start from 0 but when removing a book the index values start from 1. Please fix this issue.

### Adding task index values

![image](https://user-images.githubusercontent.com/83803414/173189489-261e33d5-38cd-4fa1-8494-b44f17811ef2.png)

### Removing index values

![image](https://user-images.githubusercontent.com/83803414/173189510-1f0bd480-5072-4c3c-a869-3c841d514272.png)